extensions/twenty/src/create-object-record.tsx

```diff
@@ -0,0 +1,127 @@
+import { Action, ActionPanel, Detail, Icon, List, showToast, Toast, useNavigation } from "@raycast/api";
+import { randomUUID } from "crypto";
+
+import twenty from "./services/TwentySDK";
+import { ObjectIcons } from "./enum/icons";
+import { OpenCreateObjectRecordForm } from "./pages";
+import { usePromise } from "@raycast/utils";
+import { useState } from "react";
+
+export default function CreateObjectRecord() {
+  const {
+    isLoading,
+    data: activeDataModels,
+    error,
+  } = usePromise(async () => {
+    const activeDataModels = await twenty.getActiveDataModels();
+
+    if (typeof activeDataModels === "string") {
+      throw new Error(activeDataModels as string);
+    }
+
+    return activeDataModels;
+  });
+
+  const [isOpenView, setIsOpenView] = useState(false);
+  const { push } = useNavigation();
+
+  if (error) {
+    return <Detail markdown={` # ERROR \n\n ${error.message}`} />;
+  }
+
+  const standardActiveModel = activeDataModels?.filter((model) => !model.isCustom);
+  const customActiveModel = activeDataModels?.filter((model) => model.isCustom);
+
+  return (
+    <List
+      isLoading={isLoading || isOpenView}
+      navigationTitle="Create Object Record"
+      searchBarPlaceholder="Search Object Record"
+    >
+      <List.Section title="Standard Objects">
+        {standardActiveModel?.map((model) => {
+          const { id, description, labelPlural, icon } = model;
+          return (
+            <List.Item
+              id={id}
+              title={labelPlural}
+              subtitle={description ?? ""}
+              actions={
+                !isOpenView ? (
+                  <ActionPanel>
+                    <Action
+                      title="Create Record"
+                      icon={Icon.List}
+                      onAction={async () => {
+                        setIsOpenView(true);
+                        const objectRecordMetadata = await twenty.getRecordFieldsForDataModel(id);
+                        if (typeof objectRecordMetadata === "string") {
+                          await showToast({
+                            style: Toast.Style.Failure,
+                            title: objectRecordMetadata,
+                          });
+                          setTimeout(() => {
+                            setIsOpenView(false);
+                          }, 500);
+                        } else {
+                          push(OpenCreateObjectRecordForm({ objectRecordMetadata }));
+                          setIsOpenView(false);
+                        }
+                      }}
```

Comment:
I would suggest another approach instead of `setTimeout` as it's needless complexity and can have unintended side-effects.

Maybe something like:
```
onAction={async () => {
                        try {
                          setIsOpenView(true);
                          const objectRecordMetadata = await twenty.getRecordFieldsForDataModel('id');
                          if (typeof objectRecordMetadata === "string") {
                            await showToast({
                              style: Toast.Style.Failure,
                              title: objectRecordMetadata,
                            });
                          } else {
                            push(OpenCreateObjectRecordForm({ objectRecordMetadata }));
                          }
                        } finally {
                          setIsOpenView(false);
                        }
                      }}
```